client-v2/src/shared/types/Action.ts

```diff
@@ -29,10 +30,16 @@ interface UpdateArticleFragmentMeta {
   };
 }
 
+// The clipboard type here is only needed for when we are inserting
+//  into a clipboard. Ideally clipboard type would only belong to the
+//  clipboard insert action because a lot of the code to handle insertions
+//  in different places in shared is cleaner to have clipboard type here
+//  as optional
```

Comment:
If there is a simpler way of doing please let me know. Not keen to start making the code to create insert actions even more complicated, I like the fact that a lot of inserting code is shared.